### PR TITLE
Add support for IAM Auth and LTv3

### PR DESCRIPTION
**Pull request by mkistler**
This PR adds support for Watson services that use IAM authentication.  It also replaces the Language Translator V2 service with Language Translator V3.

The code in this PR is not yet fully tested, but I wanted to get feedback on these changes before going further.

### CODE REVIEW

File: src/main/java/com/ibm/watson/developer_cloud/spring/boot/WatsonAutoConfiguration.java [hunk 1]
```diff
@@ -215,7 +248,9 @@ public ToneAnalyzer toneAnalyzer() {
   public VisualRecognition visualRecognition() {
     VisualRecognition service = new VisualRecognition(vrConfig.getVersionDate());
     configUrl(service, vrConfig);
-    configBasicAuth(service, vrConfig);
+    if (!configBasicAuth(service, vrConfig)) {
```
Comment (lpatino10) [inline src/main/java/com/ibm/watson/developer_cloud/spring/boot/WatsonAutoConfiguration.java]: Does this line need to check for `configApiKey()`?

**Comment by lpatino10** [comment]
Other than one comment I think it looks good so far 👍

File: src/main/java/com/ibm/watson/developer_cloud/spring/boot/WatsonAutoConfiguration.java (hunk 1 above)
Comment (mkistler) [reply src/main/java/com/ibm/watson/developer_cloud/spring/boot/WatsonAutoConfiguration.java]: Good catch.  I think it does.  And actually I think I will flip all these tests around to check for IAM first, and only when it is not provided to use the "legacy" auth.  That will be a smaller code change.

**Comment by mkistler** [comment]
Test for IAM auth has been added so I have removed the do not merge label.

**Comment by mkistler** [comment]
@lpatino10 @BarDweller I think this is ready to go.  If one of you approve I will go ahead and merge.

**Review by germanattanasio — APPROVED** [review]

**Pull request merged**